pinecone/provider/index_resource.go

```diff
@@ -326,7 +327,7 @@ func (r *IndexResource) Read(ctx context.Context, req resource.ReadRequest, resp
 
 	index, err := r.client.DescribeIndex(ctx, data.Id.ValueString())
 	if err != nil {
-		if err.Error() == fmt.Sprintf("failed to describe idx: Resource %s not found", data.Id.ValueString()) {
+		if strings.Contains(err.Error(), "not found") {
```

Comment:
I think the error check here was too specific, I was seeing weird edge cases where things weren't properly cleared out of the terraform state and I believe it was coming from here.

Reply:
This was a hack to detect this bad state. In reality, it would be nice if there was a constant error code in the `err` from the SDK so a string match wouldn't have to occur.

Reply:
Ah good point. When I added in a custom `PineconeError` type to the Go SDK I actually put a `Code` field at the top level. Maybe we could leverage that instead: https://github.com/pinecone-io/go-pinecone/blob/main/pinecone/errors.go#L6